src/NHibernate/SqlCommand/SqlSelectBuilder.cs

```diff
@@ -181,9 +187,10 @@ public SqlSelectBuilder SetHavingClause(SqlString havingSqlString)
 			return this;
 		}
 
-		public SqlSelectBuilder SetLockMode(LockMode lockMode)
+		public SqlSelectBuilder SetLockMode(LockMode lockMode, string alias)
```

Comment:
This is a binary breaking change. Changing a public method must be done in two steps: obsolete the method, add an overload. The `alias` parameter could be a bit more explicit about what it is, such as `fromTableAlias`. Since the for update can accept a list of tables, it could even be something like `nonOuterAliases`.